draft-zsf.md

```diff
@@ -0,0 +1,154 @@
+
+Consensus nodes are then required to track new per-block state:
+
+- `ZSF_BALANCE[H] : u64 [zatoshi]`
+
+The state is a single 64 bit integer (representing units of `zatoshi`) at any given block height, ``H``, representing the Sustainability Fund balance at that height, ``H``. The `ZSF_BALANCE` can be calculated using the following formula:
+
+`TOTAL ZEC TO EXIST (MAX_MONEY) - CLAIMED BLOCK SUBSIDIES OF PAST BLOCKS + SUM OF ALL ZSF DEPOSITS FROM PAST TRANSACTIONS`
+
+This formula holds for all future blocks. It is safe and semantically consistent to consider all blocks prior to the activation height to have a value of `0` in this field. This is not required by this proposal but may be convenient in designs or implementations.
+
+### `ZSF_BALANCE` Requirements
+
+- The value of `ZSF_BALANCE` SHOULD equal `0` for all blocks prior to the activation height `H`. 
+- The above described formula `TOTAL ZEC TO EXIST (MAX_MONEY) - CLAIMED BLOCK SUBSIDIES OF PAST BLOCKS + SUM OF ALL ZSF DEPOSITS FROM PAST TRANSACTIONS` MUST hold for all future blocks.
+- Future protocol changes MUST NOT reduce or redistribute the funds in the ZSF, and they may not increase the payout rate to a reasonable approximation beyond the four year half-life constraint.
+
+## `ZSF_DEPOSIT`
+
+Each transaction can dedicate some of its excess funds to the ZSF, and the remainder becomes the miner fee, with any excess miner fee/reward going to the ZSF
+
+This is achieved by adding a new field to all transactions:
+
+- `ZSF_DEPOSIT : u64 [zatoshi]`
+
+The `ZSF_BALANCE[H]` for a block at height `H` can be calculated given a value of `ZSF_BALANCE[H-1]` and the set of transactions contained in that block. First, the `ZSF_DEPOSIT[H]` is calculated based solely on `ZSF_BALANCE[H-1]`. This is subtracted from the previous block's balance to be distributed as part of the block reward. Second, the sum of all the `ZSF_DEPOSIT` fields of all transactions in the block is added to the balance.
+
+It is safe and consistent to treat older transactions using pre-Sustainability Fund formats as if they have this field implicitly present with a value of 0 where that simplifies designs or implementations.
+
+Note: this field is not generally considered an "output" because it differs from other outputs in several significant ways:
+
+- All `ZSF_DEPOSIT` fields contribute to a single global balance rather than a user-specific output state,
+- Consensus validation can account for this field by updating `ZSF_BALANCE[H]` and do not need to track any transaction field specific state after this accounting in contrast to e.g. UTXOs which must be tracked until spent.
+- This field does not contribute to the transaction graph of senders or recipients whether transparent or protected by cryptography.
+
+### `ZSF_DEPOSIT` Requirements
+
+- There MUST be only one `ZSF_DEPOSIT` field per transaction
+- ZIP 225 [3] MUST be updated to include `ZSF_DEPOSIT`. ZIP 244 MAY be updated as well.
```

Comment:
ZIP 244 must be updated to put the `ZSF_BALANCE` fields in the transaction ID, because a change in the deposit value changes the effects of a transaction.

Reply:
Actually it is ZIP 230 (#687), i.e. the proposed v6 format ZIP, that needs to be updated.

Nitpick: MUST and MAY should only be used for requirements on implementors, not for TODOs about how to modify other ZIPs (since this is not a Process ZIP).

Reply:
This requirement might be easier to read if there's one line for each ZIP that needs updating.